monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/ScanObservable.scala

```diff
@@ -68,9 +77,13 @@ class ScanObservable[A,R](
             }
 
           def onComplete(): Unit =
-            if (!isDone) {
-              isDone = true
-              out.onComplete()
+            // `onNext` might require error reporting, so we need for it
+            // to actually finish before we can try to forward `onComplete`
+            semaphore.awaitAllReleased().foreach { _ =>
+              if (!isDone) {
```

Comment:
Here you actually need to back-pressure on `lastAck` unfortunately. It is true that `onComplete` does not need to wait for the final `onNext`, but it does need to wait for the second to last `onNext`.

So picture this situation:

```scala
Observable.now(1).scan(0)(_ + _)
```

What will happen is this:

```scala
lastAck = out.onNext(0)
lastAck.syncFlatMap { case Continue => out.onNext(1) }
out.onComplete()
```

Now if you really think about it this way, notice that `onNext(1)` can be concurrent with `onComplete` (if the downstream `Observer` returns a delayed `Future`) and that should never happen. 

Also you don't need the semaphore here either.